### ISSUE TITLE
Add options to qvm-usb to attach a device based on a pattern in the description

### ISSUE THREAD

**Issue by face** [QubesOS/qubes-issues#8300]
### The problem you're addressing (if any)

My Dell dock gives different device Ids for my USB devices on every boot.  I have  a systemd boot script that uses `qvm-usb` to attach the NIC with a hard wire wire to sys-usb after the sys-ubs qube starts.     I had to modify `qvm-usb` to attach based on  the description to get it to work.     --persistent will not work due to the different device ID on every boot.

### The solution you'd like

So, I added a -m pattern  -i index to the qubes-usb command.   The index is needed if multiple matches occur.  I'm not a python coder so I had some help from chatgpt4:  https://github.com/QubesOS/qubes-core-admin-client/compare/main...face:qubes-core-admin-client:main

If that code is close, I would be happy to learn how the automated tests are run, add some, and submit a pull reqeust.  However, as I said I'm not a python coder and got help from chatgpt...so no hard feelings if our code is not used.

For example:  

BACKEND:DEVID      DESCRIPTION                                          USED BY
sys-usb:7-1.4      Realtek_USB_10_100_1000_LAN_001000001 
sys-usb:8-2.3.3.1  Yubico_YubiKey_OTP+FIDO+CCID 
sys-usb:8-2.3.3.4  Yubico_YubiKey_OTP+FIDO+CCID 
sys-usb:9-2.4      Realtek_USB_10_100_1000_LAN_111000001  

As the DEVID changes on every boot, my new options allow the following in a systemd boot script that runs after sys-usb qube boots:
```
qvm-attach sys-net -m 'Realtek.*111000001 '

```

Also I can use it for things in dom0 like:
```
qvm-attach work -m Yubi -i 0
qvm-attach personal -m Yubi -i 1
```

I'm testing my changes on my sytem and they are working.  So far the script seems backwards compatible too.

### The value to a user, and who that user might be

Any Qubes user who would like to have persistent networking on boot when their hardware gives them different device IDs for every boot.

**Comment by andrewdavidwong** [comment]
If I understand correctly, the main security concern with this approach is that a malicious device could advertise itself by a false description in an attempt to get itself automatically attached or to deceive the user into attaching it.

**Comment by face** [comment]
> If I understand correctly, the main security concern with this approach is that a malicious device could advertise itself by a false description in an attempt to get itself automatically attached or to deceive the user into attaching it.

Yes, but this is the same security concern as --persistent, which is also supported.   

Which I was actually thinking about last night....it would be nice if --match and --index worked with --persistent too.  Then I wouldn't need a the attached systemd script.   When I have time I'll look at how --persistent works internally.

**Comment by ben-grande** [comment]
Based on patterns on the description is not safe. Will close as duplicate of another issue though.
